docs/source/developer_guides/method_comparison/lora.md

```diff
@@ -0,0 +1,196 @@
+
+## Implementation
+
+### Basic Usage
+```python
+from peft import LoraConfig, get_peft_model
+
+# Define LoRA configuration
+config = LoraConfig(
+    r=8,  # rank
+    lora_alpha=32,
+    target_modules=["q_proj", "v_proj"],
+    lora_dropout=0.05,
+    bias="none",
+)
+
+# Create PEFT model
+model = get_peft_model(model, config)
+```
+
+### Advanced Configuration
+```python
+# Custom LoRA configuration for specific needs
+config = LoraConfig(
+    r=16,  # higher rank for better performance
+    lora_alpha=64,
+    target_modules=["q_proj", "k_proj", "v_proj", "o_proj"],
+    lora_dropout=0.1,
+    bias="lora_only",
+    modules_to_save=["classifier"],
+)
+```
+
+## Hyperparameter Tuning
+
+### Recommended Ranges
+| Parameter | Recommended Range | Impact |
+|-----------|------------------|--------|
+| rank (r) | 4-32 | Higher = better performance, more parameters |
+| alpha | 8-64 | Controls scaling of LoRA weights |
+| dropout | 0.0-0.1 | Regularization, prevent overfitting |
+
+### Optimal Settings by Model Size
+| Model Size | Rank | Alpha | Dropout |
+|------------|------|-------|---------|
+| < 1B      | 4-8  | 16-32 | 0.05    |
+| 1B-7B     | 8-16 | 32-64 | 0.05    |
+| 7B-13B    | 16-32| 64    | 0.1     |
+| > 13B     | 32   | 64    | 0.1     |
+
+## Comparison with Other Methods
+
+### Performance Comparison
+| Method | Memory Efficiency | Training Speed | Use Case Flexibility |
+|--------|------------------|----------------|----------------------|
+| LoRA   | High            | Fast          | High                |
+| Full FT | Low            | Slow          | High                |
+| Adapter | Medium         | Medium        | Medium              |
+| Prompt | Very High      | Very Fast     | Low                 |
+
+### Memory Usage Comparison
+| Method | Parameters (% of base) | Memory Overhead |
+|--------|----------------------|-----------------|
+| LoRA   | 0.1%                | Low            |
+| Full FT | 100%               | High           |
+| Adapter | 0.5%               | Medium         |
+| Prompt | 0.01%              | Very Low       |
+
+## Best Practices
+
+1. **Rank Selection**
+   - Start with rank 8 for most cases
+   - Increase rank for better performance if needed
+   - Consider model size when choosing rank
+
+2. **Target Modules**
+   - Include attention layers (q_proj, v_proj)
+   - Add more layers for complex tasks
+   - Consider model architecture
+
+3. **Training Tips**
+   - Use learning rate 1e-4 to 5e-4
+   - Apply gradient clipping
+   - Monitor loss convergence
+
+## Common Issues and Solutions
+
+### Problem: Slow Training
+**Solution:**
+```python
+# Optimize training speed
+config = LoraConfig(
+    r=8,
+    lora_alpha=32,
+    target_modules=["q_proj", "v_proj"],  # Focus on key layers
+    lora_dropout=0.0,  # Remove dropout for speed
+)
+```
+
+### Problem: High Memory Usage
+**Solution:**
+```python
+# Reduce memory usage
+config = LoraConfig(
+    r=4,  # Lower rank
```

Comment:
Generally, the rank will only have a small impact on overall memory usage. It's not wrong to write this, but IMO a typical user should try other steps first (quantized models, batch size, activation checkpointing, etc.)